src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,174 @@
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Packaging.Licenses;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine
+{
+
+    [Command(typeof(NuGetCommand), "search", "SearchCommandDescription",
+            UsageSummaryResourceName = "SearchCommandUsageSummary", UsageExampleResourceName = "SearchCommandUsageExamples")]
+    public class SearchCommand : Command
+    {
+        private readonly int _lineSeparatorLength = 20;
+
+        private readonly List<string> _sources = new List<string>();
+
+        [Option(typeof(NuGetCommand), "SearchCommandSourceDescription")]
+        public ICollection<string> Source => _sources;
+
+        [Option(typeof(NuGetCommand), "SearchCommandPreRelease")]
+        public bool PreRelease { get; set; } = false;
+
+        [Option(typeof(NuGetCommand), "SearchCommandTake")]
+        public int Take { get; set; } = 20;
+
+        private IList<PackageSource> GetEndpointsAsync()
+        {
+            List<PackageSource> configurationSources = SourceProvider.LoadPackageSources()
+                .Where(p => p.IsEnabled)
+                .ToList();
+
+            IList<PackageSource> packageSources;
+            if (Source.Count > 0)
+            {
+                packageSources = Source
+                    .Select(s => PackageSourceProviderExtensions.ResolveSource(configurationSources, s))
+                    .ToList();
+            }
+            else
+            {
+                packageSources = configurationSources;
+            }
+            return packageSources;
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            string sourceSeparator = new string('=', _lineSeparatorLength);
+            string packageSeparator = new string('-', _lineSeparatorLength);
+
+            ILogger logger = Console;
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            SearchFilter searchFilter = new SearchFilter(includePrerelease: PreRelease);
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
+            {
+                SourceRepository repository = Repository.Factory.GetCoreV3(source);
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+
+                if (resource is null)
+                {
+                    continue;
```

Comment:
Might be a good diea to report to the user that the feed does not support search.

Reply:
This already throws an exception to that effect:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Resources/PackageSearchResourceV3.cs#L180

Reply:
What if the feed is not a v3 feed? 
Try https://www.nuget.org/api/v2 for example.

Reply:
It seems to be able to find a search resource and search this feed without any problems, so there was no opportunity to throw that exception. Also, in general isn't Search only really supported by V3 (https://docs.microsoft.com/en-us/nuget/api/search-query-service-resource)?

Reply:
V2 has a search too. 
We have a search implementation for local feeds as well. 

You can look at all the implementation sof the abstract class PackageSearchResource. 

So I'd expect we don't get a null resource too often (if at all). 

I still think it might be a good idea to tell something to the user given that we tried but for whatever reason failed to obtain a search resource.

Reply:
Done, it will now print "Failed to obtain a search resource." and continue if the resource is ever returned as null.